### PR TITLE
Warning cleanup 2018-10-24

### PR DESCRIPTION
**Pull request by wthrowe**
### Types of changes:

- [ ] Bugfix
- [ ] New feature

### Component:

- [ ] Code
- [ ] Documentation
- [ ] Build system
- [ ] Continuous integration

### Code review checklist

- [ ] The PR passes all checks, including unit tests, `clang-tidy` and `IWYU`. For
  instructions on how to perform the CI checks locally refer to the [Dev guide
  on the Travis CI](https://spectre-code.org/travis_guide.html).
- [ ] The code is documented and the documentation renders correctly. Run `make doc`
  to generate the documentation locally into `BUILD_DIR/docs/html`. Then open
  `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).

### CODE REVIEW

File: tests/Unit/NumericalAlgorithms/Interpolation/Test_InterpolatorReceiveVolumeData.cpp [hunk 1]
```diff
@@ -292,7 +292,7 @@ SPECTRE_TEST_CASE("Unit.NumericalAlgorithms.Interpolator.ReceiveVolumeData",
 
   // Tell the interpolator how many elements there are by registering
   // each one.
-  for (const auto& element_id : element_ids) {
+  for (size_t i = 0; i < element_ids.size(); ++i) {
```
Comment (osheamonn) [inline tests/Unit/NumericalAlgorithms/Interpolation/Test_InterpolatorReceiveVolumeData.cpp:295]: What warning was this triggering?

Comment (wthrowe) [reply tests/Unit/NumericalAlgorithms/Interpolation/Test_InterpolatorReceiveVolumeData.cpp:295]: `element_id` unused

**Comment by nilsdeppe** [comment]
IWYU is unhappy

**Review by kidder — APPROVED** [review]

**Comment by kidder** [comment]
@nilsdeppe please look at

**Review by nilsdeppe — APPROVED** [review]

**Pull request merged**